hibernate-envers/src/test/java/org/hibernate/envers/test/integration/flush/ManualAuditFlush.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * Copyright (c) 2008, Red Hat Middleware LLC or third-party contributors as
+ * indicated by the @author tags or express copyright attribution
+ * statements applied by the authors.  All third-party contributions are
+ * distributed under license by Red Hat Middleware LLC.
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.hibernate.envers.test.integration.flush;
+
+import org.hibernate.FlushMode;
+import org.hibernate.envers.RevisionType;
+import org.hibernate.envers.query.AuditEntity;
+import org.hibernate.envers.test.Priority;
+import org.hibernate.envers.test.entities.StrTestEntity;
+import org.hibernate.testing.TestForIssue;
+import org.junit.Test;
+
+import javax.persistence.EntityManager;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Michal Skowronek (mskowr at o2 dot pl)
+ */
+@TestForIssue( jiraKey = "HHH-9622" )
+public class ManualAuditFlush extends AbstractFlushTest {
+	private Integer id;
+
+	public FlushMode getFlushMode() {
+		return FlushMode.MANUAL;
+	}
+
+	@Test
+	@Priority(10)
+	public void initData() {
```

Comment:
I would suggest this test be improved with a more complex series of operations like
```
// revision 1
entityManager.getTransaction().begin();
MyEntity entity = new MyEntity( "Taz" );
entityManager.persist( entity );
entityManager.flush();
entityManager.getTransaction().commit();

// revision 2
entityManager.getTransaction().begin();
entity = entityManager.find( MyEntity.class, entity.getId() );
entity.setName( "Roger Rabbit" );
entityManager.flush();
entityManager.unwrap( Session.class ).flushAuditWork(); // some proposed API
entityManager.remove( entityManager.find( MyEntity.class, entity.getId() ) );
entityManager.flush();
entityManager.getTransaction().commit();
```
While this will likely work under the `DefaultAuditStrategy` configuration; it will fail on `ValidityAuditStrategy`.